core/src/main/java/org/apache/iceberg/LocationProviders.java

```diff
@@ -108,10 +108,15 @@ public String newDataLocation(String filename) {
   static class ObjectStoreLocationProvider implements LocationProvider {
 
     private static final HashFunction HASH_FUNC = Hashing.murmur3_32_fixed();
-    private static final BaseEncoding BASE64_ENCODER = BaseEncoding.base64Url().omitPadding();
-    private static final ThreadLocal<byte[]> TEMP = ThreadLocal.withInitial(() -> new byte[4]);
+    // the starting index of the lower 20-bits of a 32-bit binary string
+    private static final int HASH_BINARY_STRING_START_INDEX = 12;
```

Comment:
Can we flip this to be to be `HASH_BINARY_STRING_BITS = 20`, since that's more applicable to what we're constructing?  The start index is confusing because it only has meaning in context to the length of what we're skipping from the bit field.

Reply:
Done